Automattic/woocommerce-payments
# Set payment method on order when setting title

Fixes #2401

#### Changes proposed in this Pull Request



This PR sets the payment method for an order during the UPE process payment flow. The initial bug was that payment details were not displaying on the admin order page when the payment was made using the Pay for Order flow with UPE. What I discovered is that when displaying the order meta data [there was a check](https://github.com/woocommerce/woocommerce/blob/164e7e495c123380f1350f97a282c86a2da8a60d/includes/admin/meta-boxes/class-wc-meta-box-order-data.php#L194) for the `_payment_method` meta data on the order. This was coming up empty for Pay for Order UPE orders. Digging in to why, I found that this is set during the [`pay_action` function](https://github.com/woocommerce/woocommerce/blob/164e7e495c123380f1350f97a282c86a2da8a60d/includes/class-wc-form-handler.php#L378). This was due to the $_POST['woocommerce_pay']  variable not being set due to the redirect nature of UPE checkouts.  

Rather than mucking with the flow and redirects, I decided to explicitly set the payment method meta data during the usual UPE process flow. Some precedence for this was found in the [`capture_terminal_payment` function](https://github.com/Automattic/woocommerce-payments/blob/76198fef593a89c2cb64b57b7a99b634e318847a/includes/admin/class-wc-rest-payments-orders-controller.php#L104). Even though this call is only needed for the Pay for Order flow, I couldn't foresee any issue with redundantly setting it every time, but would be happy to conditionalize it if there are concerns around that.





**Before**
![PaymentDetails-Before](https://user-images.githubusercontent.com/68524302/127696478-4a8ad9c9-a862-47a9-8818-edccf814a8f8.png)

**After**
![PaymentDetails-After](https://user-images.githubusercontent.com/68524302/127696486-bb7cbc42-fb8f-42f4-b5a9-f63f98178da4.png)

#### Testing instructions

* Start with a site with this PR pulled down and UPE enabled.
* As a site admin, go to WooCommerce > Orders > Add Order.
* Manually create an order by selecting a customer and item and creating the order.
* Click on the Customer Payment Page link from the newly created order details page.
* Pay for the order using UPE from the Pay for Order page.
* Check the order details page again and confirm that the status was set to Processing and the meta details include a link to the transaction as well as naming WooCommerce Payments.
* Regression test with regular checkouts using UPE and confirm that the details are still displayed correctly.

-------------------
